File: JavaScript_Basic_04_2023_SolutionsOfProblems/05.Ex_PB-JS-While-Loop/P07.Moving.js
function moving(input) {
    let volumeOfHouse = Number(input[0]) * Number(input[1]) * Number(input[2]);
    let i = 3;
    while (input[i] !== "Done" && volumeOfHouse > 0) {
        volumeOfHouse -= Number(input[i]);
        i++;
    }
    if (volumeOfHouse > 0) {
        console.log(`${volumeOfHouse} Cubic meters left.`);
    } else {
        console.log(`No more free space! You need ${-volumeOfHouse} Cubic meters more.`);
    }
}
moving(["10",
    "1",
    "2",
    "4",
    "6",
    "Done"])

File: SoftUni_JAVA_BASIC_Jan_2023_SolutionsOfProblems/P05Ex_While_Loop/P07_Moving.java
package P05Ex_While_Loop;

import java.util.Scanner;

public class P07_Moving {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int wight = Integer.parseInt(scanner.nextLine());
        int length = Integer.parseInt(scanner.nextLine());
        int height = Integer.parseInt(scanner.nextLine());
        int countPackage = wight * length * height;
        String input = "";
        while (countPackage > 0) {
            input = scanner.nextLine();
            if (input.equals("Done")) {
                break;
            }
            countPackage -= Integer.parseInt(input);
        }
        if (input.equals("Done")) {
            System.out.printf("%d Cubic meters left.", countPackage);
        } else {
            System.out.printf("No more free space! You need %d Cubic meters more.", -countPackage);
        }
    }
}
